Authentication/SignUp/View/SignupController.swift

```diff
@@ -65,62 +67,120 @@ private extension SignupController {
     }
     
     private func bindNameElements() {
         if let nameTextField = signupView.usernameTextField {
             _viewModel.name <~ nameTextField.rex_textSignal
             _viewModel.nameValidationErrors.signal.observeNext { [unowned self] errors in
-                self._delegate.signupController(self, didFailNameValidationWithErrors: errors)
+                if errors.isEmpty {
+                    self._delegate.signupControllerDidPassNameValidation(self)
+                } else {
+                    self._delegate.signupController(self, didFailNameValidationWithErrors: errors)
+                }
             }
             if let nameValidationMessageLabel = signupView.usernameValidationMessageLabel {
                 nameValidationMessageLabel.rex_text <~ _viewModel.nameValidationErrors.signal.map { $0.first ?? " " }
             }
+            nameTextField.delegate = self
         }
     }
     
     private func bindEmailElements() {
         _viewModel.email <~ signupView.emailTextField.rex_textSignal
         _viewModel.emailValidationErrors.signal.observeNext { [unowned self] errors in
-            self._delegate.signupController(self, didFailEmailValidationWithErrors: errors)
+            if errors.isEmpty {
+                self._delegate.signupControllerDidPassEmailValidation(self)
+            } else {
+                self._delegate.signupController(self, didFailEmailValidationWithErrors: errors)
+            }
         }
         if let emailValidationMessageLabel = signupView.emailValidationMessageLabel {
             emailValidationMessageLabel.rex_text <~ _viewModel.emailValidationErrors.signal.map { $0.first ?? " " }
         }
+        signupView.emailTextField.delegate = self
     }
     
     private func bindPasswordElements() {
-        _viewModel.password <~ signupView.passwordTextField.rex_textSignal
+        _viewModel.password <~ signupView.passwordTextField.rex_textSignal.on(next: { [unowned self] text in
+            if text.isEmpty {
+                self.signupView.passwordVisibilityButton?.hidden = true
+            } else {
+                self.signupView.passwordVisibilityButton?.hidden = false
+            }
+        })
         _viewModel.passwordValidationErrors.signal.observeNext { [unowned self] errors in
-            self._delegate.signupController(self, didFailPasswordValidationWithErrors: errors)
+            if errors.isEmpty {
+                self._delegate.signupControllerDidPassPasswordValidation(self)
+            } else {
+                self._delegate.signupController(self, didFailPasswordValidationWithErrors: errors)
+            }
         }
         if let passwordValidationMessageLabel = signupView.passwordValidationMessageLabel {
             passwordValidationMessageLabel.rex_text <~ _viewModel.passwordValidationErrors.signal.map { $0.first ?? " " }
```

Comment:
I think we should express this in a different way, as it is (<~ under the hood do this) observing a new signal every time nameValidationErrors emit a value (and we try to avoid it)